### DISCUSSION TITLE
Outgoing Spam Threshold

### DISCUSSION THREAD

**Discussion by SaasX** [postalserver/postal#2642] [Help with using Postal]
Hello,

I've just set up SpamAssassin to analyze outgoing emails and block messages with a poor spam score. However, I'm encountering an issue with blacklist checks:

URIBL_DBL_BLOCKED_OPENDNS

ADMINISTRATOR NOTICE: The query to dbl.spamhaus.org was blocked due to usage of an open resolver. See https://www.spamhaus.org/returnc/pub/ [URI: [www.xxxxxxx.com](http://www.xxxxxxx.com/)] [URI: xxxxxxxxcom]

URIBL_BLOCKED

ADMINISTRATOR NOTICE: The query to URIBL was blocked. See http://wiki.apache.org/spamassassin/DnsBlocklists#dnsbl-block for more information. [URI: xxxxxxxxx.com]

I have a strict SPF configuration, and I can't seem to resolve this issue. Do you have a solution?

Thank you.

**Comment by willpower232** [comment]
It looks like this is a configuration issue, either with `/etc/resolv.conf` or you could fix it within spamassassins own configuration

change your server:
https://support.cpanel.net/hc/en-us/articles/360053079473-SpamAssassin-headers-show-URIBL-Blocked-error-when-Exim-uses-RBLs
https://forums.cpanel.net/threads/query-to-uribl-was-blocked-how-do-i-set-up-a-caching-nameserver.593567/

OR

change your spamassassin configuration:
https://github.com/hestiacp/hestiacp/issues/1086#issuecomment-678661842

**Reply by SaasX** [reply]
Hello Will,

Thank you for your response. With your help, I managed to resolve one of the two error messages (URIBL_DBL_BLOCKED_OPENDNS), but I still have this one remaining:

URIBL_BLOCKED
ADMINISTRATOR NOTICE: The query to URIBL was blocked. See http://wiki.apache.org/spamassassin/DnsBlocklists#dnsbl-block for more information. [URI: xxx.com]

The first error is resolved with Google DNS in /etc/resolv.conf, but it doesn't work for the second one. Modifying the spamassassin file as in your last link completely disables the spam filter and doesn't yield any results.

**Reply by SaasX** [reply]
I add in /etc/mail/spamassassin/local.cf

header URIBL_BLOCKED eval:check_rbl('uribl', 'lastexternal', 'list.uribl.com.')
describe URIBL_BLOCKED URIBL: Contains an URL listed in the URIBL blacklist
tflags URIBL_BLOCKED net
score URIBL_BLOCKED 0 0 0 0

header SPAMHAUS_BLOCKED eval:check_rbl('zen', 'lastexternal', 'zen.spamhaus.org.')
describe SPAMHAUS_BLOCKED Spamhaus ZEN: Contains an IP listed in Spamhaus ZEN
tflags SPAMHAUS_BLOCKED net
score SPAMHAUS_BLOCKED 0 0 0 0

I no longer have any errors, does that seem correct to you?

**Reply by willpower232** [reply]
Unfortunately I am not a spamassassin expert, if it works for you then sounds like a plan. I think Postal will tell you if it fails to do any checks because the configuration doesn't work.

**Reply by SaasX** [reply]
After some testing it seems to work. Thank you for your help.

**Reply by SaasX** [reply]
A last question will, about postalhq this time. I use this rule for spamassassin : 

header URIBL_BLOCKED eval:check_rbl('uribl', 'lastexternal', 'list.uribl.com.')
describe URIBL_BLOCKED URIBL: Contains an URL listed in the URIBL blacklist
tflags URIBL_BLOCKED net
score URIBL_BLOCKED 0 0 0 0

header SPAMHAUS_BLOCKED eval:check_rbl('zen', 'lastexternal', 'zen.spamhaus.org.')
describe SPAMHAUS_BLOCKED Spamhaus ZEN: Contains an IP listed in Spamhaus ZEN
tflags SPAMHAUS_BLOCKED net
score SPAMHAUS_BLOCKED 0 0 0 0

Do you know if we can apply outgoing email verification for cloudmark under the same model? My tests do not work with cloudmark.

thx

**Reply by willpower232** [reply]
I am also not familiar with cloudmark so not really sure I'm afraid.
